### PR TITLE
Style widgets

### PR DESCRIPTION
**Pull request by merryman**
This branch will provide a complete set of graphical widgets for direct manipulation of frequently used types of values in Morphic (.i.e. Color, Gradient, Number, Enumerable, Rectangle, Point...)
The widgets will be universally embeddable, meaning they are not restricted to be used in certain contexts (such as the inspector), but can (and should) be embedded in various tools or interactive essays.
The goal of style widgets is, to make classic morph properties easily recognizable and changeable in a straight forward manner. This provides a useful building block for interactive essays or other dynamic representations of applications in lively.

### CODE REVIEW

**Comment by rksm** [comment]
Heads up: 2f7b2f5597925a6e15dad3a06ca5d61f2646c015 replaces the icon picker with an inline version as the old one took more than 3secs to load on startup.

**Comment by rksm** [comment]
Looks good, some issues I found:

- tooltips look weird:

screen shot 2017-06-15 at 11 46 43 pm

--------------

- When clicking/dragging in the positioning widget, frame gets moved
screen shot 2017-06-15 at 11 48 05 pm

--------------

- Style widget unusable
style-sheet-widget-issue

--------------

- Layout widget sets layout but doesn't update layouter name

layout-widget-issue

File: ide/js/inspector.js [hunk 1]
```diff
@@ -107,7 +111,227 @@ var defaultPropertyOptions = {
   sortFunction: (target, props) => Array.isArray(target) ? props : props.sort(defaultSort)
 }
 
-function propertiesOf(target) {
+/*
+
+When using trees as a means to simply display information to the user
```
Comment (rksm) [inline ide/js/inspector.js:116]: 👍  for the comments

File: text/morph.js [hunk 2]
```diff
@@ -188,6 +188,9 @@ export class Text extends Morph {
       },
 
       padding: {
+        spec: {
```
Comment (rksm) [inline text/morph.js]: Is it necessary to have a sub-field "spec"? Why not add type/values directly to property?

**Comment by rksm** [comment]
There seem to be a number of new entities ("popovers", controls, etc.)  Can you please make a list and provide a short description of what entities are there and what they do.

File: text/morph.js (hunk 2 above)
Comment (merryman) [reply text/morph.js]: That's a good question. I have to admit I first just put this new set of information into a separate field to simplify the implementation process, but yeah we could also just add new fields to the properties. My hunch was probably that we may wanna create type objects in the future that provide more rich interaction capabilities and these are easier to extract if all this info is already stored in a separate field.

Comment (rksm) [reply text/morph.js]: We might do that but I think for now its fine (and more user friendly) to keep it simple.

**Comment by rksm** [comment]
After collapse/uncollapse filters aren't applied
inspector-filter-issue

**Comment by rksm** [comment]
The inspector does not seem to be updating the properties shown anymore?

**Comment by rksm** [comment]
How does one get rid of the color tool in the style sheet editor?
getting-rid-of-color-palette-in-style-sheet-editor

**Pull request merged**